domain/schema/model/sql/0018-application.sql

```diff
@@ -71,26 +71,36 @@ CREATE TABLE application_endpoint_space (
 CREATE TABLE application_endpoint_cidr (
     application_uuid TEXT NOT NULL,
     cidr TEXT,
-    CONSTRAINT fk_application_endpoint_space_application
+    CONSTRAINT fk_application_endpoint_cidr_application
     FOREIGN KEY (application_uuid)
     REFERENCES application (uuid),
     PRIMARY KEY (application_uuid, cidr)
 );
 
 CREATE TABLE application_config (
     application_uuid TEXT NOT NULL,
-    name TEXT NOT NULL,
-    type_id TEXT,
+    "key" TEXT NOT NULL,
+    type_id INT NOT NULL,
```

Comment:
Surely this should have failed somewhere? How can we have a foriegn key that's a different type to the primary key it references?

Reply:
It wasn't used, so never triggered. Don't forget foreign keys are lazy by default.